lane_detector:
  ros__parameters:
    debug: False
    image_topic: "/zed/zed_node/rgb/image_rect_color"
    # image_topic: "/zed/zed_node/right/image_rect_color"
    disable_topic: "/temp_disable"

# All tests on lane 4
pure_pursuit:
  ros__parameters:
    debug: True
    drive_topic: "/vesc/high_level/input/nav_0"
    max_speed: 4.0
    low_pass_cutoff_freq: 15.0
    # low_pass_cutoff_freq: 15.0
    base_lookahead: 3.9
    lookahead_speed_ratio: 0.0
    # base_lookahead: 5.0
    # eta_multiplier: 1.5
    eta_multiplier: 1.0
    # eta_multiplier: 1.25
    # eta_multiplier: 1.4
    Kp: 0.0
    Kd: 0.1
    # Kd: 0.1

    # Works at 2.5 m/s
    # max_speed: 2.5
    # low_pass_cutoff_freq: 15.0
    # base_lookahead: 1.5
    # lookahead_speed_ratio: 0.6
    # eta_multiplier: 1.5

    # Worked before kinda at 4 m/s
    # max_speed: 4.0
    # low_pass_cutoff_freq: 40.0
    # base_lookahead: 1.5
    # lookahead_speed_ratio: 0.6
    # eta_multiplier: 1.5
    disable_topic: "/temp_disable"

visualizer_node:
  ros__parameters:
    debug: False
    fast_mode: True
    image_topic: "/zed/zed_node/rgb/image_rect_color"
    # image_topic: "/zed/zed_node/right/image_rect_color"

safety_controller:
  ros__parameters:
    scan_topic: "/scan"
    ackermann_cmd_topic: "/vesc/low_level/ackermann_cmd"
    safety_topic: "/vesc/low_level/input/safety"
    disable_topic: "/temp_disable"
    temp_disable_timeout: 0.1
    stopping_time: 0.7
